cmd/thanos/main.go

```diff
@@ -143,8 +142,26 @@ func main() {
 	{
 		ctx := context.Background()
 
-		var closeFn func() error
-		tracer, closeFn = tracing.NewOptionalGCloudTracer(ctx, logger, *gcloudTraceProject, *gcloudTraceSampleFactor, *debugName)
+		var closer io.Closer
+		var confContentYaml []byte
+		confContentYaml, err = tracingConfig.Content()
+
+		if len(confContentYaml) == 0 {
+			level.Info(logger).Log("msg", "Tracing will be disabled")
+			tracer, closer, err = provider.NoopTracer()
+		} else {
+			tracer, closer, err = provider.NewTracer(ctx, logger, confContentYaml)
+		}
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, errors.Wrapf(err, "tracing failed"))
+			if closer != nil {
```

Comment:
In this particular error case closer will be never not nil, let's remove this.

Reply:
However in any other error case below we need to close this  for example in line 184